File: CS/CustomSecurityOperation.Module/DatabaseUpdate/Updater.cs
using System;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Updating;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.BaseImpl;
using DevExpress.ExpressApp.Security;
using DevExpress.Persistent.BaseImpl.PermissionPolicy;
using DevExpress.Persistent.Base;

namespace CustomSecurityOperation.Module.DatabaseUpdate {
    public class Updater : ModuleUpdater {
        public Updater(IObjectSpace objectSpace, Version currentDBVersion) : base(objectSpace, currentDBVersion) { }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
            PermissionPolicyUser admin = ObjectSpace.FindObject<PermissionPolicyUser>(new BinaryOperator("UserName", "Admin"));
            if (admin == null) {
                admin = ObjectSpace.CreateObject<PermissionPolicyUser>();
                admin.UserName = "Admin";
                admin.SetPassword("");
                PermissionPolicyRole adminRole = ObjectSpace.CreateObject<PermissionPolicyRole>();
                adminRole.Name = "Administrator Role";
                adminRole.IsAdministrative = true;
                admin.Roles.Add(adminRole);
            }
            PermissionPolicyUser user = ObjectSpace.FindObject<PermissionPolicyUser>(new BinaryOperator("UserName", "User"));
            if (user == null) {
                user = ObjectSpace.CreateObject<PermissionPolicyUser>();
                user.UserName = "User";
                user.SetPassword("");
                PermissionPolicyRole userRole = ObjectSpace.CreateObject<PermissionPolicyRole>();
                userRole.Name = "User Role";
                CustomTypePermissionObject taskTypePermission = ObjectSpace.CreateObject<CustomTypePermissionObject>();
                taskTypePermission.TargetType = typeof(Task);
                taskTypePermission.CreateState = SecurityPermissionState.Allow;
                taskTypePermission.DeleteState = SecurityPermissionState.Allow;
                taskTypePermission.NavigateState = SecurityPermissionState.Allow;
                taskTypePermission.ReadState = SecurityPermissionState.Allow;
                taskTypePermission.WriteState = SecurityPermissionState.Allow;
                taskTypePermission.ExportState = SecurityPermissionState.Allow;
                CustomTypePermissionObject userTypePermission = ObjectSpace.CreateObject<CustomTypePermissionObject>();
                userTypePermission.TargetType = typeof(PermissionPolicyUser);
                userTypePermission.NavigateState = SecurityPermissionState.Allow;
                userTypePermission.ReadState = SecurityPermissionState.Allow;
                userRole.TypePermissions.Add(taskTypePermission);
                userRole.TypePermissions.Add(userTypePermission);
                user.Roles.Add(userRole);
            }
            ObjectSpace.CommitChanges();
            for (int i = 1; i <= 10; i++) {
                string subject = string.Format("Task {0}", i);
                Task task = ObjectSpace.FindObject<Task>(new BinaryOperator("Subject", subject));
                if (task == null) {
                    task = ObjectSpace.CreateObject<Task>();
                    task.Subject = subject;
                    task.DueDate = DateTime.Today;
                    task.Save();
                }
            }
            ObjectSpace.CommitChanges();
        }
    }
}

File: VB/CustomSecurityOperation.Module/DatabaseUpdate/Updater.vb
Imports System

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.ExpressApp.Security
Imports DevExpress.Persistent.BaseImpl.PermissionPolicy
Imports DevExpress.Persistent.Base

Namespace CustomSecurityOperation.Module.DatabaseUpdate
	Public Class Updater
		Inherits ModuleUpdater

		Public Sub New(ByVal objectSpace As IObjectSpace, ByVal currentDBVersion As Version)
			MyBase.New(objectSpace, currentDBVersion)
		End Sub
		Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
			MyBase.UpdateDatabaseAfterUpdateSchema()
			Dim admin As PermissionPolicyUser = ObjectSpace.FindObject(Of PermissionPolicyUser)(New BinaryOperator("UserName", "Admin"))
			If admin Is Nothing Then
				admin = ObjectSpace.CreateObject(Of PermissionPolicyUser)()
				admin.UserName = "Admin"
				admin.SetPassword("")
				Dim adminRole As PermissionPolicyRole = ObjectSpace.CreateObject(Of PermissionPolicyRole)()
				adminRole.Name = "Administrator Role"
				adminRole.IsAdministrative = True
				admin.Roles.Add(adminRole)
			End If
			Dim user As PermissionPolicyUser = ObjectSpace.FindObject(Of PermissionPolicyUser)(New BinaryOperator("UserName", "User"))
			If user Is Nothing Then
				user = ObjectSpace.CreateObject(Of PermissionPolicyUser)()
				user.UserName = "User"
				user.SetPassword("")
				Dim userRole As PermissionPolicyRole = ObjectSpace.CreateObject(Of PermissionPolicyRole)()
				userRole.Name = "User Role"
				Dim taskTypePermission As CustomTypePermissionObject = ObjectSpace.CreateObject(Of CustomTypePermissionObject)()
				taskTypePermission.TargetType = GetType(Task)
				taskTypePermission.CreateState = SecurityPermissionState.Allow
				taskTypePermission.DeleteState = SecurityPermissionState.Allow
				taskTypePermission.NavigateState = SecurityPermissionState.Allow
				taskTypePermission.ReadState = SecurityPermissionState.Allow
				taskTypePermission.WriteState = SecurityPermissionState.Allow
				taskTypePermission.ExportState = SecurityPermissionState.Allow
				Dim userTypePermission As CustomTypePermissionObject = ObjectSpace.CreateObject(Of CustomTypePermissionObject)()
				userTypePermission.TargetType = GetType(PermissionPolicyUser)
				userTypePermission.NavigateState = SecurityPermissionState.Allow
				userTypePermission.ReadState = SecurityPermissionState.Allow
				userRole.TypePermissions.Add(taskTypePermission)
				userRole.TypePermissions.Add(userTypePermission)
				user.Roles.Add(userRole)
			End If
			ObjectSpace.CommitChanges()
			For i As Integer = 1 To 10
				Dim subject As String = String.Format("Task {0}", i)
				Dim task As Task = ObjectSpace.FindObject(Of Task)(New BinaryOperator("Subject", subject))
				If task Is Nothing Then
					task = ObjectSpace.CreateObject(Of Task)()
					task.Subject = subject
					task.DueDate = Date.Today
					task.Save()
				End If
			Next i
			ObjectSpace.CommitChanges()
		End Sub
	End Class
End Namespace
